### PR TITLE
(SR-9248) check #tfop result types

### PR DESCRIPTION
**Pull request by marcrasi**
This PR checks that tfop result types conform to TensorGroup, or that they are tuples of types that conform to TensorGroup.

This resolves https://bugs.swift.org/browse/SR-9248, and this should unblock https://github.com/apple/swift/pull/20591.

### CODE REVIEW

File: lib/Sema/CSApply.cpp [hunk 1]
```diff
@@ -2443,82 +2443,111 @@ namespace {
...
-        auto argType = cs.getType(argExpr);
-
-        // CSGen has already ensured that the first argument is a string, so skip
-        // the first argument.
-        if (i == 0) {
-          assert(argType->isEqual(stringTy) &&
-                 "first argument should be string");
-          continue;
-        }
+      } else {
```
Comment (marcrasi) [inline lib/Sema/CSApply.cpp]: fyi, all the changes until L2521 are just putting existing code in the else block and indenting it

Comment (mhong) [reply lib/Sema/CSApply.cpp]: Thanks for the helpful note!

**Review by mhong — APPROVED** [review]
Very nice patch, and thanks for getting the other PR unblocked!

FYI @rxwei

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Review by rxwei — APPROVED** [review]

**Pull request merged**